src/Identity/Endpoints/src/IdentityBearerAuthenticationHandler.cs

```diff
@@ -0,0 +1,111 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity.Endpoints.DTO;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Identity.Endpoints;
+
+internal sealed class IdentityBearerAuthenticationHandler : SignInAuthenticationHandler<IdentityBearerAuthenticationOptions>
+{
+    private const string BearerTokenPurpose = $"Microsoft.AspNetCore.Identity.Endpoints.IdentityBearerAuthenticationHandler:v1:BearerToken";
+
+    private static readonly Task<AuthenticateResult> TokenMissingTask = Task.FromResult(AuthenticateResult.Fail("Token missing"));
+    private static readonly Task<AuthenticateResult> FailedUnprotectingTokenTask = Task.FromResult(AuthenticateResult.Fail("Unprotect token failed"));
+    private static readonly Task<AuthenticateResult> TokenExpiredTask = Task.FromResult(AuthenticateResult.Fail("Token expired"));
+
+    private readonly IDataProtectionProvider _fallbackDataProtectionProvider;
+
+    public IdentityBearerAuthenticationHandler(
+        IOptionsMonitor<IdentityBearerAuthenticationOptions> optionsMonitor,
+        ILoggerFactory loggerFactory,
+        UrlEncoder urlEncoder,
+        ISystemClock clock,
+        IDataProtectionProvider dataProtectionProvider)
+        : base(optionsMonitor, loggerFactory, urlEncoder, clock)
+    {
+        _fallbackDataProtectionProvider = dataProtectionProvider;
+    }
+
+    private IDataProtectionProvider DataProtectionProvider
+        => Options.DataProtectionProvider ?? _fallbackDataProtectionProvider;
+
+    private ISecureDataFormat<AuthenticationTicket> BearerTokenProtector
+        => Options.BearerTokenProtector ?? new TicketDataFormat(DataProtectionProvider.CreateProtector(BearerTokenPurpose));
+
+    protected override Task<AuthenticateResult> HandleAuthenticateAsync()
+    {
+        // If there's no bearer token, forward to cookie auth.
+        if (GetBearerTokenOrNull() is not string token)
+        {
+            return Options.BearerTokenMissingFallbackScheme is string fallbackScheme
+                ? Context.AuthenticateAsync(fallbackScheme)
+                : TokenMissingTask;
+        }
```

Comment:
We need to enable getting this token from the query string for things like SignalR. We should design a callback on the options similar to the JWT Bearer auth handler.